### PR TITLE
Add elfeed layer in contrib/tools

### PR DESCRIPTION
**Pull request by usharf**
Add elfeed (atom and rss feed reader) support layer with some customizations.

### CODE REVIEW

**Comment by robbyoconnor** [comment]
:+1:

**Comment by usharf** [comment]
@darkfeline wasn't sure how to do it... i.e. so that users can set configuration options up with :variables rather than straight forward set etc.

File: contrib/tools/elfeed/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+### Custom Key Bindings
+
+Key Binding        | Description
+-------------------|-------------
+<kbd>q</kbd>       | quit main window, or item view buffer.
+<kbd>c</kbd>       | compact feed db 
+<kbd>o</kbd>       | looad OPML
+<kbd>w</kbd>       | start web server
+<kbd>W</kbd>       | stop web server
+
```
Comment (robbyoconnor) [inline contrib/tools/elfeed/README.md]: this should be in org not markdown.

**Comment by darkfeline** [comment]
@usharf Do we have a policy for doing that?  Because I don't see using `setq` as a bad thing, Spacemacs shouldn't be trying to replace Emacs entirely, just provide useful defaults.

I guess there's no harm leaving it there.  Actually, I take that back.  Either you define all of elfeed's variables here, or you define none of them.  Otherwise, the user experience is inconsistent: "Why can I use `:variables` to configure this elfeed option but not that one?"

**Comment by syl20bnr** [comment]
Indeed if the layer variable is only an alias to a stock variable then the layer variable is useless. See #1990 or https://github.com/syl20bnr/spacemacs/pull/1480#discussion_r30383116 for examples of this.

**Comment by usharf** [comment]
@syl20bnr @darkfeline that's what I was struggling with. It's somewhat inconsistent.  I support the idea the spacemacs should not "replicate" things and keeping most seeing straight forward will also make it easy to migrate configurations between installation, weather using spacemacs or not, though from a usability point-of-view, as well as discoverability, having :variables be a one-stop-shop for all configuration settings (at least the meaningful ones) is tempting. 

Will re-write the code.

**Comment by syl20bnr** [comment]
Same thing for this layer, I cannot merge it, the repository is unknwon.
Can you re-submit it ? I'll try to merge it more quickly. Thank you !

**Comment by d12frosted** [comment]
@usharf any update on this?

**Comment by usharf** [comment]
@d12frosted better close, I won't have time to check this again.
10x

**Comment by d12frosted** [comment]
@usharf Ok, since I am using elfeed, I can work on this in very near future.

**Comment by d12frosted** [comment]
Anyway, thanks for bringing it here! :smile:

**Pull request closed without merging**